engine/access/ingestion2/latest_persisted_sealed_result.go

```diff
@@ -0,0 +1,112 @@
+package ingestion2
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+)
+
+// LatestPersistedSealedResult tracks the most recently persisted sealed execution result processed
+// by the ingestion engine.
+type LatestPersistedSealedResult struct {
+	// resultID is the execution result ID of the most recently persisted sealed result.
+	resultID flow.Identifier
+
+	// height is the height of the most recently persisted sealed result's block.
+	// This is the value stored in the consumer progress index.
+	height uint64
+
+	// cp is the consumer progress instance
+	cp storage.ConsumerProgress
+
+	// writeMu is used to prevent concurrent batch updates to the persisted height.
+	// the critical section is fairly large, so use a separate mutex from the cached values.
```

Comment:
I don't quite understand why we need additional lock, maybe some benchmark could show the difference? I thought `mu.RLock` should be relatively cheap, even if `mu.Lock` takes a long time to be unlocked, no?

Reply:
we need to prevent concurrent calls to `BatchSet`, so this needs to lock for the duration of the batch creation and commit logic.

there may be reads happening at the same time, but those only need to protect the actual writes to the cached state.

maybe I could rename them `batchMu` and `cacheMu` for clarity?

Reply:
> I thought mu.RLock should be relatively cheap, even if mu.Lock takes a long time to be unlocked, no?

my understanding is that if the full lock is held, all attempts to get the read lock will block until it is released.